sql/core/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -374,67 +374,29 @@ object SchemaConverters extends Logging {
     }
   }
 
-  private def getDefaultValue(dataType: DataType): Any = {
-    def createNestedDefault(st: StructType): java.util.HashMap[String, Any] = {
-      val defaultMap = new java.util.HashMap[String, Any]()
-      st.fields.foreach { field =>
-        field.dataType match {
-          case nested: StructType =>
-            // For nested structs, recursively create the default structure
-            defaultMap.put(field.name, createNestedDefault(nested))
-          case _ =>
-            // For leaf fields, use null
-            defaultMap.put(field.name, null)
-        }
+  private def createDefaultStruct(st: StructType): java.util.HashMap[String, Any] = {
+    val defaultMap = new java.util.HashMap[String, Any]()
+    st.fields.foreach { field =>
+      field.dataType match {
+        case nested: StructType =>
+          defaultMap.put(field.name, createDefaultStruct(nested))
+        case _ =>
+          defaultMap.put(field.name, null)
       }
-      defaultMap
-    }
-
-    dataType match {
-      // Basic types
-      case BooleanType => null
-      case ByteType | ShortType | IntegerType => null
-      case LongType => null
-      case FloatType => null
-      case DoubleType => null
-      case StringType => null
-      case BinaryType => null
-
-      // Complex types
-      case ArrayType(_, _) => new java.util.ArrayList[Any]()
-      case MapType(StringType, _, _) => new java.util.HashMap[String, Any]()
-      case st: StructType => createNestedDefault(st)
-
-      // Special types
-      case _: DecimalType => java.nio.ByteBuffer.allocate(0)
-      case DateType => null
-      case TimestampType => null
-      case TimestampNTZType => null
-      case NullType => null
-      case _ => null
     }
+    defaultMap
   }
 
   def toAvroTypeWithDefaults(
```

Comment:
No, I'm not only concerned about nullable. If you revisit callers of this method, I don't see any parameter except catalystType to be used, and catalystType is always StructType. Shall we distinguish the method we are exposing to caller vs the method to recursively resolve?